src/request_body_processor/xml.cc

```diff
@@ -25,11 +25,122 @@ namespace RequestBodyProcessor {
 
 #ifdef WITH_LIBXML2
 
+/*
+* NodeData for parsing XML into args
+*/
+NodeData::NodeData() {
+    has_child = false;
+}
+
+NodeData::~NodeData() {};
+
+/*
+* XMLNodes for parsing XML into args
+*/
+XMLNodes::XMLNodes(Transaction *transaction) 
+    : nodes{},
+    node_depth(0),
+    currpath(""),
+    currval(""),
+    m_transaction(transaction)
+    {}
+
+XMLNodes::~XMLNodes() {};
+
+/*
+* SAX handler for parsing XML into args
+*/
+class MSCSAXHandler {
+    public:
+        void onStartElement(void * ctx, const xmlChar *localname) {
+
+            std::string name = reinterpret_cast<const char*>(localname);
+
+            XMLNodes* xml_data = static_cast<XMLNodes*>(ctx);
+            xml_data->nodes.push_back(std::make_shared<NodeData>());
+            xml_data->node_depth++;
+            // FIXME - later if we want to check the depth of XML tree
+            /* if (max_depth > 0 && max_depth > xml_data->node_depth) {
+                std::cout << "Depth of XML tree reached the given maximum value " << xml_data->node_depth << std::endl;
+                exit(1);
+            } */
+            // if it's not the first (root) item, then append a '.'
+            // note, this can't occur because there is always a pseudo root element: 'xml'
```

Comment:
By "this", do you mean that the condition should always be true because of the pseudo root element? If so, I suggest this edit:
```suggestion
            // note, the condition should always be true because there is always a pseudo root element: 'xml'
```

Reply:
Indeed you're right, thanks.